app/src/main/java/sdp/moneyrun/Game.java

```diff
@@ -2,36 +2,61 @@
 
 import android.location.Location;
 
+
+
+
+import java.util.Arrays;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
 
 // The entirety of the game logic should be implemented in this class
 public class Game {
+    private final String gameId;
+    private final String name;
     private List<Player> players;
+    private int maxPlayerCount;
     private List<Riddle> riddles;
-    private Location startLocation;//TODO: check if we will use the existing or create a new class Location
+    private Location startLocation; //TODO: check if we will use the existing or create a new class Location
     private List<Coin> coins;
 
-    public Game(List<Player> players, List<Riddle> riddles, List<Coin> coins, Location startLocation) {
-        if (players == null || riddles == null || startLocation == null || players.isEmpty() || riddles.isEmpty())
+    public Game(String gameId, String name, List<Player> players, int maxPlayerCount, List<Riddle> riddles, List<Coin> coins, Location startLocation) {
+        if (players == null || riddles == null || startLocation == null)
             throw new IllegalArgumentException("Null parameter passed as argument in Game constructor");
-
+        this.gameId = gameId;
+        this.name = name;
         this.players = players;
+        this.maxPlayerCount = maxPlayerCount;
         this.riddles = riddles;
         this.startLocation = startLocation;
         this.coins = coins;
     }
 
-    public static void startGame(Game game) {
-        game.startGame();
+    public String getGameId(){
+        return gameId;
     }
 
-    // Launched when create game button is pressed
-    public void startGame() {
+    public String getName(){
```

Comment:
Very easy and quick tests that would improve coverage.

Reply:
done